Common/cpp/Tools/JSISerializer.cpp

```diff
@@ -0,0 +1,200 @@
+    if (i != propsCount - 1) {
+      ss << ", ";
+    }
+  }
+
+  ss << '}';
+
+  return ss.str();
+}
+
+std::string stringifyJSError(jsi::Runtime &rt, const jsi::Object &object) {
+  std::stringstream ss;
+  ss << '[' << object.getProperty(rt, "name").toString(rt).utf8(rt) << ": "
+     << object.getProperty(rt, "message").toString(rt).utf8(rt) << ']';
+
+  return ss.str();
+}
+
+std::string stringifyJSSet(jsi::Runtime &rt, const jsi::Object &object) {
+  std::stringstream ss;
+  jsi::Function arrayFrom = rt.global()
+                                .getPropertyAsObject(rt, "Array")
+                                .getPropertyAsFunction(rt, "from");
+  jsi::Object result = arrayFrom.call(rt, object).asObject(rt);
+
+  if (!result.isArray(rt)) {
+    return "[Set]";
+  }
+
+  ss << "Set {" << reanimated::stringifyJSIValue(rt, result.asArray(rt)) << '}';
+
+  return ss.str();
+}
+
+std::string stringifyJSMap(jsi::Runtime &rt, const jsi::Object &object) {
+  std::stringstream ss;
+  jsi::Function arrayFrom = rt.global()
+                                .getPropertyAsObject(rt, "Array")
+                                .getPropertyAsFunction(rt, "from");
+  jsi::Object result = arrayFrom.call(rt, object).asObject(rt);
+
+  if (!result.isArray(rt)) {
+    return "[Map]";
+  }
+
+  auto arr = result.asArray(rt);
+  auto length = arr.size(rt);
+
+  ss << "Map {";
+
+  for (size_t i = 0; i < length; i++) {
+    auto pair = arr.getValueAtIndex(rt, i).asObject(rt).getArray(rt);
+    auto key = pair.getValueAtIndex(rt, 0);
+    auto value = pair.getValueAtIndex(rt, 1);
+    ss << reanimated::stringifyJSIValue(rt, key) << ": "
+       << reanimated::stringifyJSIValue(rt, value);
+    if (i != length - 1) {
+      ss << ", ";
+    }
+  }
+
+  ss << '}';
+
+  return ss.str();
+}
+
+std::string reanimated::stringifyJSIValue(
+    jsi::Runtime &rt,
+    const jsi::Value &value) {
+  if (value.isBool() || value.isNumber()) {
+    return value.toString(rt).utf8(rt);
+  } else if (value.isString()) {
+    return '"' + value.getString(rt).utf8(rt) + '"';
+  } else if (value.isSymbol()) {
+    return value.getSymbol(rt).toString(rt);
+  } else if (value.isBigInt()) {
+    return value.getBigInt(rt).toString(rt).utf8(rt) + 'n';
+  } else if (value.isUndefined()) {
+    return "undefined";
+  } else if (value.isNull()) {
+    return "null";
+  } else if (value.isObject()) {
+    jsi::Object object = value.asObject(rt);
+
+    if (object.isArray(rt)) {
+      return stringifyJSIArray(rt, object.getArray(rt));
+    } else if (object.isArrayBuffer(rt)) {
+      return stringifyJSIArrayBuffer(rt, object.getArrayBuffer(rt));
+    } else if (object.isFunction(rt)) {
+      return stringifyJSIFunction(rt, object.getFunction(rt));
+    } else if (object.isHostObject(rt)) {
+      return stringifyJSIHostObject(rt, *object.asHostObject(rt).get());
+    } else if (getPrototypeName(rt, object) == "[object Map]") {
```

Comment:
I'm afraid this can differ between JS runtimes. Do you think there's any way to get only `"Map"`? Maybe we could compare if the prototype object is equal to `Object.getPrototypeOf(new Map())` or do the same thing as we do for `Error` below?